src/mysky/index.ts

```diff
@@ -682,14 +682,21 @@ export class MySky {
 
     // Add DAC permissions.
     const perms = dac.getPermissions();
-    this.addPermissions(...perms);
+    await this.addPermissions(...perms);
   }
 
-  protected handleLogin(loggedIn: boolean): void {
+  protected async handleLogin(loggedIn: boolean): Promise<void> {
     if (loggedIn) {
-      for (const dac of this.dacs) {
-        dac.onUserLogin();
-      }
+      await Promise.all(
+        this.dacs.map(async (dac) => {
+          try {
+            await dac.onUserLogin();
+          } catch (error) {
+            // Don't throw on error, just print a console warning.
+            console.warn(error);
+          }
+        })
+      );
```

Comment:
use `allSettled` and you won't need to do that artificial try-catch

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled

Reply:
I had to set the target to `es2020` and then I was getting lint failures so I had to set `moduleResolution: node`, and then CI failed for Node 12. Unless we want to drop support for users of Node 12, we can't use `Promise.allSettled`, so I reverted the commit.